src/advanced_agent/database/connection.py

```
@@ -0,0 +1,295 @@
+                return False
+            
+            # 現在のデータベースをバックアップ
+            import time
+            current_backup = f"{self.db_path}.backup.{int(time.time())}"
+            if os.path.exists(self.db_path):
+                shutil.copy2(self.db_path, current_backup)
+            
+            # バックアップから復元
+            shutil.copy2(backup_path, self.db_path)
+            logger.info(f"データベース復元完了: {backup_path}")
+            return True
```

Ruff (0.12.2)

Consider moving this statement to an `else` block

(TRY300)